LayoutTests/css3/filters/resources/filter-helpers.js

```diff
@@ -0,0 +1,210 @@
+        (0.272 - 0.272 * oneMinusAmount), (0.534 - 0.534 * oneMinusAmount), (0.131 + 0.869 * oneMinusAmount), 0, 0,
+        0, 0, 0, 1, 0        
+    ];
+
+    return elementFromHTMLString(`<feColorMatrix type="matrix" values="${values.join(' ')}"/>`);
+}
+
+function svgSaturateComponent(filterValue)
+{
+    return elementFromHTMLString(`<feColorMatrix type="saturate" values="${filterValue}"/>`);
+}
+
+function svgHueRotateComponent(filterValue)
+{
+    const valuePattern = /(.+)turn/;
+    const matchResult = valuePattern.exec(filterValue);
+    if (!matchResult) {
+        console.log('hue-rotate only accepts angles in turns');
+        return null;
+    }
+    const angleInDegrees = 360 * matchResult[1];
+    
+    return elementFromHTMLString(`<feColorMatrix type="hueRotate" values="${angleInDegrees}"/>`);
+}
+
+function svgInvertComponent(filterValue)
+{
+    const oneMinusAmount = 1 - filterValue;
+    return elementFromHTMLString(`<feComponentTransfer>
+          <feFuncR type="table" tableValues="${filterValue} ${oneMinusAmount}"/>
+          <feFuncG type="table" tableValues="${filterValue} ${oneMinusAmount}"/>
+          <feFuncB type="table" tableValues="${filterValue} ${oneMinusAmount}"/>
+        </feComponentTransfer>`);
+}
+
+function svgOpacityComponent(filterValue)
+{
+    const oneMinusAmount = 1 - filterValue;
+    return elementFromHTMLString(`<feComponentTransfer>
+          <feFuncA type="table" tableValues="0 ${filterValue}"/>
+      </feComponentTransfer>`);
+}
+
+function svgBlurComponent(filterValue)
+{
+    const valueRegExp = /(.+)px/;
+    const matchResult = valueRegExp.exec(filterValue);
+    if (!matchResult) {
+        console.log('blur only accepts px units');
+        return null;
+    }
+    const stdDeviation = matchResult[1];
+    
+    return elementFromHTMLString(`<feGaussianBlur stdDeviation="${stdDeviation}"/>`);
+}
+
+function svgDropShadowComponent(filterValue)
+{
+    // drop-shadow() parameters are (x offset, y offset, blur, color).
+    const valuePattern = /(.+)\s+(.+)\s+(.+)\s+(.+)/;
+    const matchResult = valuePattern.exec(filterValue);
+    if (!matchResult) {
+        console.log(`Failed to match drop-shadow parameters ${filterValue}`);
+        return null;
+    }
+    
+    const xOffset = matchResult[1].replace(/(px)?$/, '');
+    const yOffset = matchResult[2].replace(/(px)?$/, '');
+    const stdDeviation = matchResult[3].replace(/(px)?$/, '');
+    const color = matchResult[4];
+
+    return elementFromHTMLString(`<filter>
+        <feGaussianBlur in="SourceAlpha" stdDeviation="${stdDeviation}"/>
+          <feOffset dx="${xOffset}" dy="${yOffset}" result="offsetblur"/>
+          <feFlood flood-color="${color}"/>
+          <feComposite in2="offsetblur" operator="in"/>
+          <feMerge>
+            <feMergeNode/>
+            <feMergeNode in="SourceGraphic"/>
+          </feMerge>
```

Comment:
Is there a reason we can't use 

`<feDropShadow dx="${xOffset}" dy="${yOffset}" stdDeviation="${stdDeviation}" flood-color="${color}"/>`

Also I think the effects starting from <feOffset> are indented one extra tab.